web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -292,41 +292,11 @@ private void setApplicationDataDirectory(ServletContext servletContext) {
 	}
 	
 	/**
-	 * Hacky way to get the current contextPath. This will usually be "openmrs". This method will be
-	 * obsolete when servlet api ~2.6 comes out...at which point a call like
-	 * servletContext.getContextRoot() would be sufficient
-	 *
 	 * @return current contextPath of this webapp without initial slash
 	 */
 	private String getContextPath(ServletContext servletContext) {
 		// Get the context path without the request.
-		String contextPath = "";
-		try {
-			contextPath = servletContext.getContextPath();
-		}
-		catch (NoSuchMethodError ex) {
-			// ServletContext.getContextPath() was added in version 2.5 of the Servlet API. Tomcat 5.5
-			// has version 2.4 of the servlet API, so we fall back to the hacky code we were previously
-			// using
-			try {
-				String path = servletContext.getResource("/").getPath();
-				contextPath = path.substring(0, path.lastIndexOf("/"));
-				contextPath = contextPath.substring(contextPath.lastIndexOf("/"));
-			}
-			catch (Exception e) {
-				log.error("Failed to get construct context path", e);
-			}
-		}
-		catch (Exception e) {
-			log.error("Failed to get context path", e);
-		}
-		
-		// trim off initial slash if it exists
-		if (contextPath.contains("/")) {
-			contextPath = contextPath.substring(1);
-		}
```

Comment:
@HelioStrike , thank you on this work. Are you however sure this is unlikely to happen?

Reply:
Thank you for reviewing the code. I didn't what you meant by "Are you however sure this is unlikely to happen?". The tests all seem to pass. I tested things by running `mvn clean install` a few times, so I guess things are good to go.

Reply:
Ohh, I see now! I've updated the code. Please have another look. :)